### ISSUE TITLE
errno: 150 "Foreign key constraint is incorrectly formed"

### ISSUE THREAD

**Issue by arvins-wittymanager** [bizley/yii2-migration#54]
I think there is a bug. After I run php yii migration/create-all, the tables are created in alphabetical order. However, the foreign key constraints are added inside the table creation which is in the line after the this->createTable something like this:

```
$this->createTable('foo', [ 'id' => $this->primaryKey() ]); // whatever
$this->addForeignKey('fk_app_assignee_user-id', '{{%appointment_assignee}}', 'user_id', '{{%users}}', 'id', 'RESTRICT', 'RESTRICT');
```

where table appointment_assignee is created and trying to add a foreign key constraint referencing the users table but it doesn't exist yet.

I believe one Laravel migration generator tool solves this by first creating migration files for create table commands, then creating migration files for foreign key constraint commands afterwards.

**Comment by bizley** [comment]
Yes, that's how it works now (bad, I know!) - create all does not take FKs order into consideration. I have planned to add it in next version.

**Comment by bizley** [comment]
Fixed in [3.4.0](https://github.com/bizley/yii2-migration/releases/tag/3.4.0)/[2.7.0](https://github.com/bizley/yii2-migration/releases/tag/2.7.0).

Adding the same feature to the updating migrations require refactoring the code so I'm planning to release that as 4.0.0 version.
